yoti-sdk-auth/src/main/java/com/yoti/auth/AuthenticationTokenGenerator.java

```diff
@@ -0,0 +1,166 @@
+     * @throws ResourceException if something was incorrect with the request to the Yoti authentication service
+     * @throws IOException
+     */
+    public CreateAuthenticationTokenResponse generate(List<String> scopes) throws ResourceException, IOException {
+        String jwts = signedJwtGenerator.create(sdkId, keyPair, jwtIdSupplier, authApiUrl);
+
+        Map<String, String> formParams = new HashMap<>(DEFAULT_FORM_VALUES);
+        formParams.put("scope", String.join(" ", scopes));
+        formParams.put("client_assertion", jwts);
+
+        String responseBody = formRequestClient.performRequest(authApiUrl, "POST", formParams);
+
+        return objectMapper.readValue(responseBody, CreateAuthenticationTokenResponse.class);
+    }
+
+    public static final class Builder {
+
+        private String sdkId;
+        private KeyPairSource keyPairSource;
+        private Supplier<String> jwtIdSupplier = () -> UUID.randomUUID().toString();
+
+        private Builder() {}
+
+        /**
+         * Sets the SDK ID that the authorization token will be generated against.
+         *
+         * @param sdkId the SDK ID
+         * @return the builder for method chaining.
+         */
+        public Builder withSdkId(String sdkId) {
+            this.sdkId = sdkId;
+            return this;
+        }
+
+        /**
+         * Sets the {@link KeyPairSource} that will be used to load the {@link KeyPair}
+         *
+         * @param keyPairSource the key pair source that will be used to load the {@link KeyPair}
+         * @return the builder for method chaining.
+         */
+        public Builder withKeyPairSource(KeyPairSource keyPairSource) {
+            this.keyPairSource = keyPairSource;
+            return this;
+        }
+
+        /**
+         * Sets the supplier that will be used to generate a unique ID for the
+         * authorization token.  By default, this will be a UUID v4.
+         *
+         * @param jwtIdSupplier the supplier used for generating authorization token ID
+         * @return the builder for method chaining.
+         */
+        public Builder withJwtIdSupplier(Supplier<String> jwtIdSupplier) {
+            this.jwtIdSupplier = jwtIdSupplier;
+            return this;
+        }
+
+        /**
+         * Builds an {@link AuthenticationTokenGenerator} using the values supplied to the {@link Builder}.
+         *
+         * @return the configured {@link AuthenticationTokenGenerator}
+         */
+        public AuthenticationTokenGenerator build() {
+            notNullOrEmpty(sdkId, "sdkId");
+            notNull(keyPairSource, "keyPairSource");
+            notNull(jwtIdSupplier, "jwtIdSupplier");
+
+            ObjectMapper objectMapper = new ObjectMapper()
+                    .setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE)
+                    .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

Comment:
If we do not expose the `ObjectMapper` then there's no point in making it an instance var. It should be declared as a static final var of `AuthenticationTokenGenerator`